<a href="https://colab.research.google.com/github/tjain858/Google-play-EDA/blob/main/Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

##**INTRODUCTION :**
The objective of this project is to deliver insights to understand customer demands better and thus help developers to popularize the product.It is of 10k Play Store apps for analyzing the Android market. It consists of in total of 10841 rows and 13 columns. The tools that are going to be used for this EDA would be numpy, pandas, matplotlib and seaborn which I have learnt from the course..

**Imports**: Let us start by importing some of the required libraries with which we will be working on.

In [ ]:
#imports Some importance libries
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

## **Data Prepration and Cleaning:** In this section, we will be loading the Google Store Apps data stored in csv using pandas which is a fast and powerful python library for data analysis and easy data manipulation in pandas DataFrame object. We will then perform some data preparation and also cleaning on it.

* ### Load the file using pandas
* ### Look at the some information about the data & the columns
* ### Fix any missing or incorrect values

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
Dataset_url = 'https://drive.google.com/drive/folders/1O6uEhqbxf71Bk-8xz8kdKeD-Tb2YNq_d'

In [ ]:
# Loading the dataset as a pandas data frame.
playstore_df = pd.read_csv('/content/drive/MyDrive/CAPSTON PROJECT EDA/Google-play-EDA/Play Store Data.csv')

After loading the dataset, we can start the exploration but before that, we need to check and see that the dataset is ready for performing several exploration operations or not, so let’s first have a look at the structure and the manner in which the data is *organized*

In [ ]:
# look at the first 10 records in the apps dataframe
playstore_df.head(10)


App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                         Paper flowers instructions  ART_AND_DESIGN     4.4   
6            Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   
7                                   Infinite Painter  ART_AND_DESIGN     4.1   
8                               Garden Coloring Book  ART_AND_DESIGN     4.4   
9                      Kids Paint Free - Drawing Fun  ART_AND_DESIGN     4.7   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   
5     167  5.6M      50,000+  Free     0       Everyone   
6     178   19M      50,000+  Free     0       Everyone   
7   36815   29M   1,000,000+  Free     0       Everyone   
8   13791   33M   1,000,000+  Free     0       Everyone   
9     121  3.1M      10,000+  Free     0       Everyone   

                      Genres        Last Updated         Current Ver  \
0               Art & Design     January 7, 2018               1.0.0   
1  Art & Design;Pretend Play    January 15, 2018               2.0.0   
2               Art & Design      August 1, 2018               1.2.4   
3               Art & Design        June 8, 2018  Varies with device   
4    Art & Design;Creativity       June 20, 2018                 1.1   
5               Art & Design      March 26, 2017                 1.0   
6               Art & Design      April 26, 2018                 1.1   
7               Art & Design       June 14, 2018            6.1.61.1   
8               Art & Design  September 20, 2017               2.9.2   
9    Art & Design;Creativity        July 3, 2018                 2.8   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up  
5    2.3 and up  
6  4.0.3 and up  
7    4.2 and up  
8    3.0 and up  
9  4.0.3 and up

In [ ]:
playstore_df.tail()

App             Category  \
10836                               Sya9a Maroc - FR               FAMILY   
10837               Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10836       Everyone          Education     July 25, 2017                1.48   
10837       Everyone          Education      July 6, 2018                 1.0   
10838       Everyone            Medical  January 20, 2017                 1.0   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10836          4.1 and up  
10837          4.1 and up  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

In [ ]:
# look at the random 10 records in the apps dataframe
playstore_df.sample(10)

App            Category  \
5833                                Ay Sabz Gunbad Waly       VIDEO_PLAYERS   
5196   AI Today : Artificial Intelligence News & AI 101  NEWS_AND_MAGAZINES   
2297                               Hospitalist Handbook             MEDICAL   
9176                                          EB Mobile              FAMILY   
5487  AP Statistics Prep: Practice Tests and Flashcards              FAMILY   
8953                           US DV Lottery 2020 Apply    TRAVEL_AND_LOCAL   
5216                     Turbo AI™ - Personal assistant        PRODUCTIVITY   
6868                                            BV MAAp            BUSINESS   
3318                      Free & Premium VPN - FinchVPN               TOOLS   
963                                                Nick       ENTERTAINMENT   

      Rating Reviews  Size     Installs  Type   Price Content Rating  \
5833     NaN       4  3.9M       1,000+  Free       0       Everyone   
5196     5.0      43  2.3M         100+  Free       0       Everyone   
2297     4.8      12   18M       1,000+  Paid  $19.99       Everyone   
9176     1.7    1172  5.6M      10,000+  Free       0       Everyone   
5487     4.2      19  5.4M       5,000+  Free       0       Everyone   
8953     4.6      41  2.0M      10,000+  Free       0       Everyone   
5216     2.6      91  2.8M       5,000+  Free       0       Everyone   
6868     NaN       9  4.2M         100+  Free       0       Everyone   
3318     4.2   19096   10M   1,000,000+  Free       0       Everyone   
963      4.2  123279   25M  10,000,000+  Free       0   Everyone 10+   

                           Genres        Last Updated Current Ver  \
5833      Video Players & Editors  September 11, 2017         1.1   
5196             News & Magazines       June 22, 2018         1.0   
2297                      Medical        June 5, 2017       6.0.4   
9176                    Education     October 9, 2017       1.1.2   
5487                    Education        July 3, 2018     1.6.7.1   
8953               Travel & Local    January 14, 2018        1.02   
5216                 Productivity       July 16, 2018      0.0.52   
6868                     Business    December 4, 2015       1.1.3   
3318                        Tools        July 5, 2018       2.0.2   
963   Entertainment;Music & Video    January 24, 2018       2.0.8   

       Android Ver  
5833    4.0 and up  
5196    4.4 and up  
2297    4.1 and up  
9176    4.1 and up  
5487    4.1 and up  
8953    4.0 and up  
5216  4.0.3 and up  
6868    4.1 and up  
3318    4.1 and up  
963     4.4 and up

In [ ]:
# TO know checking what columns are there in the dataset
playstore_df.columns


Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [ ]:
# look at the number of rows and columns in dataset
playstore_df.shape

(10841, 13)

In [ ]:
# look at the info of the dataframe
playstore_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


#**Description of App Dataset columns**

1.    App : The name of the app
2.    Category : The category of the app
3.    Rating : The rating of the app in the Play Store
4.    Reviews : The number of reviews of the app
5.    Size : The size of the app
6.    Install : The number of installs of the app
7.    Type : The type of the app (Free/Paid)
8.    price: The price of the app (0 if it is Free)
9.    Content Rating :The appropiate target audience of the app
10.   Genres: The genre of the app
11.   Last Updated : The date when the app was last updated
12.   Current Ver : The current version of the app
13.   Android Ver : The minimum Android version required to run the app

In [ ]:
# To check missing vlaues per columns 
playstore_df.isna().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

We saw that the dataset contains many Null or missing values. The column Rating, Type , Content Rating , Current Ver , and Android Ver contains . 1474, 1, 1, 8, and 3 missing values respectively.

Will it not be better if we can define a function to get more useful information about the different attributes of the dataset, also there is one more valid point in defining a function which it will be reusable, and we are going to utilize our defined function several times in future


In [ ]:
# Define a function to get check null vlaues in dataset
def printinfo():

    '''This function return the data type , null values and unique value in dataframe
  '''
    temp = pd.DataFrame(index=playstore_df.columns)
    temp['data_type'] = playstore_df.dtypes
    temp['null_count'] = playstore_df.isnull().sum()
    temp['unique_count'] = playstore_df.nunique()
    return temp

In [ ]:
printinfo()

data_type  null_count  unique_count
App               object           0          9660
Category          object           0            34
Rating           float64        1474            40
Reviews           object           0          6002
Size              object           0           462
Installs          object           0            22
Type              object           1             3
Price             object           0            93
Content Rating    object           1             6
Genres            object           0           120
Last Updated      object           0          1378
Current Ver       object           8          2832
Android Ver       object           3            33